### DISCUSSION TITLE
怎么单独校验动态表单项

### DISCUSSION THREAD

**Discussion by iambinlin** [ant-design/ant-design#34667] [Q&A] [answered]
```
form.validateFields(['xxx'])
```
好像没用，校验竟然通过了，进了 `then`函数

**Comment by yoyo837** [comment]
上链接，我这边正常的。

**Reply by iambinlin** [reply]
稍等，我写个demo到`sandbox`

**Reply by iambinlin** [reply]
https://codesandbox.io/s/dong-tai-zeng-jian-qian-tao-zi-duan-antd-4-19-3-forked-bztdyf?file=/index.js 
@yoyo837  看下，应该是必现。不知道是不是我使用方式有问题

**Comment by yoyo837** [comment, marked as answer]
users 不是一个具体字段。

**Reply by iambinlin** [reply]
但是如果我新增了5个动态增减项，我需要一次校验这5个，有没有什么办法。`demo`里没有这个场景

**Reply by iambinlin** [reply]
换句话说，就是我需要单独校验一整个动态增减表单域（users）。

**Reply by yoyo837** [reply]
试一下 https://codesandbox.io/s/dong-tai-zeng-jian-qian-tao-zi-duan-antd-4-19-3-forked-nu9sjo?file=/index.js:417-584

**Reply by yoyo837** [reply]
其他的我还没学会，😂，我还在3.x上。

**Reply by iambinlin** [reply]
> 试一下 https://codesandbox.io/s/dong-tai-zeng-jian-qian-tao-zi-duan-antd-4-19-3-forked-nu9sjo?file=/index.js:417-584

好吧。可以了，感谢！

但是如果要校验10个以上动态增减表单项（在一个动态增减域下）。。。那不要累死了，希望官方大大提供个便捷的方案～

**Reply by yoyo837** [reply]
通过onValueChange得到动态部分的长度，然后 form.validateFields() 参数里拼出合适的参数。

**Reply by iambinlin** [reply]
> 通过onValueChange得到动态部分的长度，然后 form.validateFields() 参数里拼出合适的参数。

好的，这个可行，非常感谢～🥳

**Comment by iambinlin** [comment]
我把`google`、`issue`、`disscussions`翻烂了都没找到有人和我碰到一样的问题。（： 手动😢

**Comment by iambinlin** [comment]
<img width="811" alt="image" src="https://user-images.githubusercontent.com/39019913/159716882-afc38554-d18d-409f-ae32-0251f2d3c84d.png">

留个坑位，琢磨了下，这么做应该也可以，集中过滤出没值的表单项，生成校验字段域～
